soda_machine.rb

```diff
@@ -1,18 +1,25 @@
 class SodaMachine
   attr_reader :sodas, :cash
+  attr_accessor :sold_soda
 
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
+    @sold_soda = @brand
   end
 
   def current_inventory_count
+    @sodas.count
   end
 
   def find_soda(soda_brand)
+    i = @sodas.index(soda_brand)
```

Comment:
Try to always keep in mind what data types you are dealing with. This would work in `@sodas` was an array of strings, but it is actually an array of `Soda` objects and so it will never match up with the `soda_brand` that is passed in (which is a `String`).